### PR TITLE
add backoff logic to agent syncs

### LINKED ISSUE #1241: agent request backoff
Related area as #1240 

On sync / fetch failure agent should gradually backoff requests. This avoids continuous doomed retries

### PR DESCRIPTION
**Pull request by amoore877**
Signed-off-by: Andrew Moore <andrew.s.moore@uber.com>

**Pull Request check list**

- [X] Commit conforms to CONTRIBUTING.md?
- [X] Proper tests/regressions included?
- [X] Documentation updated?

**Affected functionality**
Agent Manager `Fetch` and `Rotate` cadence.

**Description of change**
Bring in new open source `backoff` logic library, use for regular Agent `Fetch` and `Rotate` logic (independently). On errors in such an operation, exponentially back off up to 24 times the original interval (for context, 5 second interval would extend up to 2 minutes). Backoff logic explicitly never fails out from too many / too long of retries.
On successful operation, reset backoff to original interval.

Not quite sure ATM where / how to put some tests. Tests against the backoff library seem redundant given the original library has pretty thorough unit testing. Tests of `Manager` and `Rotator` don't seem to include possibility of called client returning an error.

**Which issue this PR fixes**
Addresses main concern of #1241 , but does not address potential aspect of ensuring backoff does not go beyond the TTL of any SVIDs. That may be worth a different Issue and PR.

### CODE REVIEW

File: pkg/agent/manager/manager.go [hunk 1]
```diff
@@ -10,8 +10,10 @@ import (
 	"time"
 
 	"github.com/andres-erbsen/clock"
+	"github.com/cenkalti/backoff/v3"
 	observer "github.com/imkira/go-observer"
 	"github.com/spiffe/spire/pkg/agent/client"
+	spire_backoff "github.com/spiffe/spire/pkg/agent/common/backoff"
```
Comment (azdagron) [inline pkg/agent/manager/manager.go]: wonder if we could avoid doing named imports if we type aliased `backoff.Backoff` in the `.../agent/common/backoff` package?

Comment (amoore877) [reply pkg/agent/manager/manager.go]: ah good call :)

File: pkg/agent/common/backoff/backoff.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import (
+	"time"
+
+	"github.com/andres-erbsen/clock"
+	v3backoff "github.com/cenkalti/backoff/v3"
+)
+
+// BackOff type alias of "github.com/cenkalti/backoff/v3" BackOff, for
+// better readability in importing
+type BackOff v3backoff.BackOff
```
Comment (azdagron) [inline pkg/agent/common/backoff/backoff.go]: This is creating a new type. Type alias syntax is:

```
type BackOff = v3backoff.BackOff
```

File: pkg/agent/common/backoff/backoff.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
+package backoff
+
+import (
+	"time"
+
+	"github.com/andres-erbsen/clock"
+	v3backoff "github.com/cenkalti/backoff/v3"
```
Comment (azdagron) [inline pkg/agent/common/backoff/backoff.go]: does this import need to be named?

**Comment by azdagron** [comment]
I think is ready to go in after the conflict is resolved.

**Pull request merged**